### DISCUSSION TITLE
at least API level 30 for Android

### DISCUSSION THREAD

**Discussion by vahost** [quasarframework/quasar#12938] [CLI - Capacitor / Cordova modes] [answered]
I'm using Capacitor to build my app for Android. When I go to Google Play to upload the .apk file, I get an error, "Your app currently targets API level 29 and must target at least API level 30 to ensure it is built to the latest API optimized for security and performance. Change your app's target API level to at least 30."

Where do I change the API level in my code?

**Comment by vahost** [comment, marked as answer]
Whups! Never mind, my mistake. I had to install the latest SDKs in Android Studio.

**Comment by Daniel123454321** [comment]
At least with Cordova it worked.
I've updated Cordova to v11 and Gradle to v7.4.2. I've also installed JDK 18. And I had to remove all Android SDKs < 31.
After all that it worked like always before.
